analyzer/src/main/kotlin/curation/FilePackageCurationProvider.kt

```diff
@@ -36,15 +37,35 @@ import org.ossreviewtoolkit.utils.log
 class FilePackageCurationProvider(curationFiles: Collection<File>) : PackageCurationProvider {
     constructor(curationFile: File) : this(listOf(curationFile))
 
-    internal val packageCurations by lazy {
-        curationFiles.mapNotNull { curationFile ->
+    companion object {
+        fun from(file: File? = null, dir: File? = null): FilePackageCurationProvider {
+            val curationFiles = mutableListOf<File>()
+            file?.takeIf { it.isFile }?.let { curationFiles += it }
+            dir?.let { curationFiles += FileFormat.findFilesWithKnownExtensions(it) }
+
+            return FilePackageCurationProvider(curationFiles)
+        }
+    }
+
+    val packageCurations = run {
```

Comment:
I'd prefer explicit type `packageCurations: Set<PackageCurations>`, to make it obvious.